client.lua

```diff
@@ -163,199 +199,106 @@ local function startTestDriveTimer(testDriveTime, prevCoords)
     if not Config.UsingTarget then
         for i = 1, #Config.Shops[shopName]['ShowroomVehicles'] do
-            zones[#zones + 1] = BoxZone:Create(
-                vector3(Config.Shops[shopName]['ShowroomVehicles'][i]['coords'].x,Config.Shops[shopName]['ShowroomVehicles'][i]['coords'].y,Config.Shops[shopName]['ShowroomVehicles'][i]['coords'].z),
-                Config.Shops[shopName]['Zone']['size'],
-                Config.Shops[shopName]['Zone']['size'],
-                {
-                    name = "box_zone_" .. shopName .. "_" .. i,
-                    minZ = Config.Shops[shopName]['Zone']['minZ'],
-                    maxZ = Config.Shops[shopName]['Zone']['maxZ'],
-                    debugPoly = false,
-                })
+            local vehData = Config.Shops[shopName]['ShowroomVehicles'][i]
+            zones[#zones + 1] = lib.zones.box({
+                coords = vec3(vehData.coords.x, vehData.coords.y, vehData.coords.z),
+                size = Config.Shops[shopName]['Zone']['size'],
+                rotation = vehData.coords.w,
+                debug = Config.Shops[shopName]['Zone']['debug'],
+                onEnter = enteringVehicleSellZone,
+                inside = insideVehicleSellZone,
+                onExit = exitingVehicleSellZone
+            })
         end
-        local combo = ComboZone:Create(zones, { name = "vehCombo", debugPoly = false })
-        combo:onPlayerInOut(function(isPointInside)
-            if isPointInside then
-                if PlayerData and PlayerData.job and(PlayerData.job.name == Config.Shops[insideShop]['Job'] or Config.Shops[insideShop]['Job'] == 'none') then
-                    lib.showContext('veh_header_menu')
-                end
-            else
-                lib.hideContext()
-            end
-        end)
     else
-        local options = {
-            {
-                name = 'vehicleshop:showVehicleOptions',
-                event = 'qb-vehicleshop:client:showVehOptions',
-                icon = "fas fa-car",
-                label = Lang:t('general.vehinteraction'),
-                canInteract = function()
-                    local closestShop = insideShop
-                    return closestShop and (Config.Shops[closestShop]['Job'] == 'none' or PlayerData.job.name == Config.Shops[closestShop]['Job'])
-                end
-            }
-        }
-
-        exports.ox_target:addEntity(entity, options)
+        exports.ox_target:addLocalEntity(entity, { {
+            name = 'vehicleshop:showVehicleOptions',
+            icon = "fas fa-car",
+            label = Lang:t('general.vehinteraction'),
+            onSelect = function()
+                openVehicleSellMenu()
+            end
+        } })
     end
 end
 
--- Zones
-function createFreeUseShop(shopShape, name)
-    local zone = PolyZone:Create(shopShape, {
-        name = name,
-        minZ = shopShape.minZ,
-        maxZ = shopShape.maxZ
-    })
+local function enterShop(self)
```

Comment:
It's unclear what type of object self is. Consider adding a function hearing here to clear the ambiguity up https://github.com/sumneko/lua-language-server/wiki/Annotations

Reply:
I understand however it is not clear what should i write there. Self is an object that is clear.
---@param self object ?

Reply:
table, not object